pkg/plugin/plugin.go

```diff
@@ -24,15 +25,21 @@ type plugin struct {
 var _ api.Plugin = &plugin{}
 
 // NewPlugin creates a new plugin instance
-func NewPlugin(entry *logrus.Entry, pluginConfig *api.PluginConfig) api.Plugin {
+func NewPlugin(entry *logrus.Entry, pluginConfig *api.PluginConfig) (api.Plugin, error) {
```

Comment:
PluginConfig is becoming complex and we should be validating it and returning errors based on that.